drivers/i3c/i3c_renesas_imx3112.c

```diff
@@ -0,0 +1,416 @@
+	}
+
+	/*
+	 * Run a SETAASA on bus to switch to I3C mode. If the mux is already in I3C mode
+	 * this CCC will be ignored
+	 */
+	err = i3c_ccc_do_setaasa(cfg->bus);
+	if (err) {
+		LOG_ERR("Cannot put mux %s in I3C mode", dev->name);
+		return err;
+	}
+
+	/* Perform HW configuration for the mux based on devicetree defaults */
+	intf_cfg_val |=
+		FIELD_PREP(RF_LOCAL_INF_PULLUP_CONF, cfg->hw_config.has_external_resistor ? 1 : 0);
+	intf_cfg_val |= FIELD_PREP(RF_LOCAL_INF_IO_LEVEL, cfg->hw_config.io_voltage);
+	err = i3c_renesas_imx3112_write_reg(
+		dev, offsetof(struct i3c_renesas_imx3112_registers, local_interface_cfg),
+		intf_cfg_val);
+	if (err) {
+		LOG_ERR("Mux %s HW configuration failed", dev->name);
+		return err;
+	}
+
+	res_cfg_val |= FIELD_PREP(RF_LSCL_0_PU_RES, cfg->hw_config.ch_resistance[0].scl);
+	res_cfg_val |= FIELD_PREP(RF_LSDA_0_PU_RES, cfg->hw_config.ch_resistance[0].sda);
+	res_cfg_val |= FIELD_PREP(RF_LSCL_1_PU_RES, cfg->hw_config.ch_resistance[1].scl);
+	res_cfg_val |= FIELD_PREP(RF_LSDA_1_PU_RES, cfg->hw_config.ch_resistance[1].sda);
+	err = i3c_renesas_imx3112_write_reg(
+		dev, offsetof(struct i3c_renesas_imx3112_registers, pullup_resistor_config),
+		res_cfg_val);
+
+	if (err) {
+		LOG_ERR("Mux %s HW configuration failed", dev->name);
+		return 0;
+	}
+
+	LOG_DBG("Mux %s initalization complete", dev->name);
+	return 0;
+}
+
+static const struct i3c_driver_api imx3112_api_funcs = {
+	.i3c_xfers = i3c_renesas_imx3112_transfer,
+	.i3c_device_find = i3c_renesas_imx3112_device_find,
+	.configure = i3c_renesas_imx3112_configure,
+};
+
+#define DT_DRV_COMPAT renesas_imx3112_i3c
+#define IMX3112_CHANNEL_INIT(parent_inst, node_id, ch_num)                                         \
+	BUILD_ASSERT(DT_REG_ADDR(node_id) < NUM_CHANNELS);					   \
+	static struct i3c_device_desc i3c_renesas_imx3112_dev_arr_##parent_inst##_##ch_num[] =     \
+		I3C_DEVICE_ARRAY_DT(node_id);							   \
+	static const struct i3c_renesas_imx3112_channel_config                                     \
+		imx3112_channel_##parent_inst##_##ch_num##_config = {                              \
+			.channel_mask = BIT(DT_REG_ADDR(node_id)),                                 \
+			.mux = DEVICE_DT_GET(DT_PARENT(node_id)),                                  \
+			.device_list.i3c = i3c_renesas_imx3112_dev_arr_##parent_inst##_##ch_num,   \
+			.device_list.num_i3c =                                                     \
+				ARRAY_SIZE(i3c_renesas_imx3112_dev_arr_##parent_inst##_##ch_num),  \
+	};                                                                                         \
+	DEVICE_DT_DEFINE(node_id, i3c_renesas_imx3112_channel_initialize, NULL, NULL,              \
+			 &imx3112_channel_##parent_inst##_##ch_num##_config, POST_KERNEL,          \
+			 CONFIG_I3C_RENESAS_CHANNEL_PRIORITY, &imx3112_api_funcs);
+
+#define CHECK_MUX_CONFIG(n)									   \
+	COND_CODE_1(DT_INST_PROP_OR(n, has_external_resistor, false),                              \
```

Comment:
I don't think you need the _OR ones for booleans. They're either present and true, or absent and false.